test/cpp/lazy/test_misc.cpp

```diff
@@ -11,9 +11,29 @@ template <typename T>
 void test_hash_repeatable_sensitive(T example_a, T example_b) {
   // repeatable
   EXPECT_EQ(Hash(example_a), Hash(example_a));
+  EXPECT_EQ(MHash(example_a), MHash(example_a));
+  EXPECT_EQ(MHash(example_a, example_a), MHash(example_a, example_a));
 
   // sensitive
   EXPECT_NE(Hash(example_a), Hash(example_b));
+  EXPECT_NE(MHash(example_a), MHash(example_b));
+  EXPECT_NE(MHash(example_a, example_a), MHash(example_a, example_b));
+}
+
+TEST(HashTest, Scalar) {
+  c10::Scalar a(0);
+  c10::Scalar b(0);
+
+  // simulate some garbage in the unused bits of the
+  // the tagged union that is c10::Scalar, which is bigger
+  // than the size of the int64_t we're currently using it with
+  *((uint8_t*)&b)  = 1;
```

Comment:
Can we do `static_cast<uint8_t*>` here? Can you explain the memory layout here a bit such that it's easier for me to understand why writing to "offset 0" makes sense? Is there any way we can tell the differences from `a` to `b` such that we can also verify that this operation succeeds?

Reply:
There isn't any way to verify it without violating the API of the Scalar.  I didn't know the memory layout either, except by experimenting.  I found it's sizeof() was 32 bytes, which surprised me.  I don't know exactly what memory i'm stepping on here, but, for the purposes of this experiment it's not important as long as it isn't memory used by the `Long` member field, which it isn't.

I did verify it though, by observing that the below EXPECTs all failed when I clobber this memory without fixing the hash function.

I don't see the point of changing the cast.  I don't benefit from any compiler safety checks here, in fact, i'm doing something 'unsafe' on purpose.

Reply:
Just for you, I decided to try this :) 
But it doesn't appear to work:
`Static_cast from 'c10::Scalar *' to 'uint8_t *' (aka 'unsigned char *') is not allowed`

Anyway, regular cast is fine.

Reply:
Yea, my bad. It should be reinterpret_cast for your use case:
`*(reinterpret_cast<uint8_t*>(&b))  = 1;`.

Reply:
Did a little research. It seems non-trivial to find any resources to answer me the memory layout question. So, it's fine to keep the comment intact.